### ISSUE TITLE
Mediator

### ISSUE THREAD

**Issue by SquidDev** [SquidDev-CC/Howl#15]
The environment and options hooks are ugly.

I'm thinking a better way would be to use [Mediator Lua](http://olivinelabs.com/mediator_lua/). This is a simplistic event system. So instead we might do something like:

```lua
Mediator:subscribe({"HowlFile", "env"}, function(env)
  env.Something = MyModule
end)
```

I think this would also be useful for the Combiner module. Since #14, this is getting complicated. Rearranging it so it uses events in separate modules might be the nicer option. We could have `Combiner:StartFile`, `Combiner:StartModule`, etc...

**Comment by SquidDev** [comment]
I've done the core work in 7ab9b8004c4f2612adbf61c04c4dd3433948094b.

Combiner will be the tricky bit.

**Comment by SquidDev** [comment]
Combiner implemented in 5543458f2f60c1c67730c02c47ddb8ebfc20381d

This fires 4 events. `self` is the dependencies object. `outputFile` looks like:
```lua
outputFile = {
  write = function(contents, file = nil) end
  path = resultPath
}
```

- `Combiner:start(self, outputFile, options)` This is fired then the combiner is started. This can be used to set options, write text, etc...
- `Combiner:write(self, name, contents, options)` This is fired after a line is written to the document. `name` is the name of the string written, `contents` is the contents of the file.
 - `Combiner:include(self, name, contents, options)` This is fired before a file is 'included' This applies to whole code segments - the original files. Returning false will cause stop the entire process.
 - `Combiner:end(self, outputFile, options)` Called when the combiner finished. Used to close handles, or write messages.
